stepsizers.go

```diff
@@ -115,3 +118,61 @@ func (q *QuadraticStepSize) StepSize(l Location, dir []float64) (stepSize float6
 	copy(q.xPrev, l.X)
 	return stepSize
 }
+
+// FirstOrderStepSize estimates the initial line search step size based on the
+// assumption that the first-order change in the function will be the same as
+// that obtained at the previous iteration. That is, the initial step size s^0_k
+// is chosen so that
+//   s^0_k ∇f_k⋅p_k = s_{k-1} ∇f_{k-1}⋅p_{k-1}
+// This is useful for line search methods that do not produce well-scaled
+// descent directions, such as gradient descent or conjugate gradient methods.
+type FirstOrderStepSize struct {
+	// The step size at the first iteration is estimated as InitialStepFactor / |g|_∞.
```

Comment:
This is correct, we want to take a small step when the gradient is large (when it is small, the step size is bound by MaxStepSize).